scripts/addFromConfig.py

```diff
@@ -89,13 +84,12 @@ def _keysCheck(conf, section):
     else:
         section_ex = section
     keys = expected_keyword[section_ex]
-    optional = optional_keyword[section_ex]
     for k in keys:
-        if k.startswith(('required_input', 'optional_input')) \
-           or k in optional:
+        if k.startswith('required_input') or k.startswith('optional_input'):
             continue
-        if k not in conf[section]:
-            raise ValueError('Required key: "{0}" was not found in [{1}] section'.format(k, section))
+        else:
+            if k not in conf[section]:
+                raise ValueError('Required key: "{0}" was not found in [{1}] section'.format(k, section))
```

Comment:
Why remove the optional arguments and make them all required?

Reply:
That line did not work for me and I had raise ValueError.  
```
         if k.startswith(('required_input', 'optional_input')) \ 
```

I remember looking in all the code for "required_input",  I don' think its used.

Reply:
That's odd; I'll look into that value error. I think it's a bytes/unicode mismatch (which means there should be a test on there, too.) We definitely do use required_input and optional_input in the config files: it's whether an input product is necessary for the process to run, or just used if present.

But that's separate from the question of why this change makes all keywords for all sections in the config file mandatory, instead of some being optional. The codedir, inspectordir, and errordir used to be optional (they were inferred from the mission directory.)

Reply:
I've verified...the .startswith(tuple) line is tested in the unit testing, and I can't get it to raise a ValueError on Python 2 or 3. I'm going to put that back to the way it was and also restore the optional keywords...if we can find a problem after that, we can work to write a test that catches it, and fix it.